content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+
+Here is where we need to define some concepts related to functions:
+
+- There's a drop-down that contains the technologies that will host our functions. As mentioned previously, we will use *Azure Functions v2 (.NET Core)*.
+- Functions also provide automatic ways of triggering our code, some of which communicate with other Azure Services (Queues triggers, IoT Triggers, Blob Triggers, etc.). We'll use the [**HTTP trigger**](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook?tabs=csharp) as we want our function to trigger whenever we enter a page.
+- Authorization level: Anonymous. We don't need fancy ways of doing authentication for this so we'll leave it at that for simplicity.
+ 
+ After creating the project, we are received with boilerplate code that does some simple processing of a request; where if the HTTP request contains a name parameter, the response will be a greeting, and otherwise the request will fail with an error message:
+
+
+```csharp
+[FunctionName("Function1")]
+    public static async Task<IActionResult> Run(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+        ILogger log)
+    {
+        log.LogInformation("C# HTTP trigger function processed a request.");
+        string name = req.Query["name"];
+        string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+        dynamic data = JsonConvert.DeserializeObject(requestBody);
+        name = name ?? data?.name;
+        return name != null
+            ? (ActionResult)new OkObjectResult($"Hello, {name}")
+            : new BadRequestObjectResult("Please pass a name on the query string or in the request body");
+    } 
+```
+
+From here we can also see that some of the settings that we chose when creating the project defined the attribute of the **Run()** method (which is the main code we want to run).
+
+There are some other files in the project that we'll look into later.
+
+If we build and run the project, a console window opens up and gives us some information about our locally-hosted service:
+
+```bash
+Application started. Press Ctrl+C to shut down.
+
+Http Functions:
+
+        Function1: [GET,POST] http://localhost:7071/api/Function1
+```
+
+If we enter that URL into a browser without a name parameter, it will ask us to pass a parameter name. If we do so (for instance, by entering *http://localhost:7071/api/Function1?name=nacho*), it will say hello pertinently.
+
+We can go ahead and make our first changes to this. We should start by naming the function something more specific. For this we can change the [*FunctionName()*] attribute. I will name it *GetPageViewCount*. We would also like it to respond only to **POST** requests, so I will remove the *GET* parameter in the [*HttpTrigger()*] attribute. Lastly, we can change our code to return 0 as text for now. This is what it will look like:
```

Comment:
Usually code  chunks inside your paragraphs are rendered with `[FunctionName(MyFunc)]`

Reply:
`0` seems a weird reponse jajajaja, I figure you are trying to return something other than a `HTTP 200` to say the request was ok, how about simply returning `ok` or `{success: true}`

Reply:
That does return an HTTP 200, but the idea of returning 0 was to have a default (fake/incomplete) 'view count'.

Reply:
hm right, didnt catch that, that's ok then